File: src/main/java/com/summer/lijiahao/utils/openapi/OpenApiView.java
package com.summer.lijiahao.utils.openapi;

import com.intellij.openapi.actionSystem.ActionManager;
import com.intellij.openapi.actionSystem.ActionToolbar;
import com.intellij.openapi.actionSystem.DefaultActionGroup;
import com.intellij.openapi.ui.SimpleToolWindowPanel;
import com.summer.lijiahao.utils.openapi.actions.FormatRequestJsonAction;
import com.summer.lijiahao.utils.openapi.actions.SendRequestAction;
import com.summer.lijiahao.utils.openapi.actions.SettingAction;
import com.summer.lijiahao.utils.openapi.ui.OpenApiTool;

public class OpenApiView
        extends SimpleToolWindowPanel {
    private OpenApiTool openApiTool;

    public OpenApiView() {
        super(false, true);
        this.openApiTool = new OpenApiTool();

        DefaultActionGroup barGroup = new DefaultActionGroup();
        barGroup.add(new SendRequestAction(this.openApiTool));
        barGroup.add(new SettingAction());
        barGroup.add(new FormatRequestJsonAction(this.openApiTool));

        ActionToolbar toolbar = ActionManager.getInstance().createActionToolbar("toolbar", barGroup, true);
        toolbar.setTargetComponent(this);
        setToolbar(toolbar.getComponent());
        setContent(this.openApiTool.getComponent());
    }


}

File: src/main/kotlin/com/summer/lijiahao/nccdevtool/opanapi/OpenApiView.kt
package com.summer.lijiahao.nccdevtool.opanapi

import com.intellij.openapi.actionSystem.ActionManager
import com.intellij.openapi.actionSystem.DefaultActionGroup
import com.intellij.openapi.ui.SimpleToolWindowPanel
import com.summer.lijiahao.nccdevtool.opanapi.action.FormatRequestJsonAction
import com.summer.lijiahao.nccdevtool.opanapi.action.SendRequestAction
import com.summer.lijiahao.nccdevtool.opanapi.action.SettingAction
import com.summer.lijiahao.nccdevtool.opanapi.ui.OpenApiTool

class OpenApiView : SimpleToolWindowPanel(false, true) {
    private val openApiTool: OpenApiTool = OpenApiTool()

    init {
        val barGroup = DefaultActionGroup()
        barGroup.add(SendRequestAction(openApiTool))
        barGroup.add(SettingAction())
        barGroup.add(FormatRequestJsonAction(openApiTool))
        val toolbar = ActionManager.getInstance().createActionToolbar("toolbar", barGroup, true)
        toolbar.targetComponent = this
        setToolbar(toolbar.component)
        openApiTool.contentPane?.let { setContent(it) }
    }
}